FE-feat-createAccountbookPage-ui

JunYoung7 opened the pull request:
### 📕 Issue Number

Close #64 
<br/>

### 📗 작업 내역

> 구현 내용 및 작업 했던 내역

- Views
   - CreateAccountbookPage
![image](https://user-images.githubusercontent.com/61405355/100755166-3a3b3c80-342f-11eb-95cb-3198c1ff667d.png)

- Organisms
   - CreateAccountbookFormBox
![image](https://user-images.githubusercontent.com/61405355/100755336-6d7dcb80-342f-11eb-8e50-8aafc453a0ed.png)
   - CreateAccountbookSetting
![image](https://user-images.githubusercontent.com/61405355/100755398-7bcbe780-342f-11eb-9748-092a1a052929.png)

- Molecules
   - InviteAccountbookCard
![image](https://user-images.githubusercontent.com/61405355/100755851-f85ec600-342f-11eb-88d0-103cc2affb49.png)

   - ChangeAccountbookColorCard
![image](https://user-images.githubusercontent.com/61405355/100755900-0280c480-3430-11eb-97af-95c7950d464e.png)

- TopNav에서 backgroundColor를 Props로 넘길 수 있도록 리팩토링

- themes color primary에 purple 추가
<br/>

### 📘 작업 유형

- 신규 기능 추가
- 리펙토링

<br/>

### 📝 PR 특이 사항

> PR을 볼 때 주의깊게 봐야하거나 말하고 싶은 점

- 가계부 생성 페이지 UI 틀 작성했습니다. 수정, 요구사항 피드백은 언제나 환영입니다!!
- TopNav organisms에 대해서 interface Props로 backgroundColor를 선택적으로 넘길 수 있도록 리팩토링 했습니다. 확인해주시고 나중에 사용하시면 될 것 같습니다.

<br/><br/>

JunYoung7 committed:
feat : CreateAccountbook 구현

- 가계부 생성 입력 부분을 구현하기 위한 CreateAccountBook 구현
- 상단부 topnav와의 높이 조절을 위한 margin-top을 2rem으로 맞춤

JunYoung7 committed:
feat : CreateAccountbookSetting 파일 생성

- + CreateAccountbookFormBox 오타 수정
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
feat : ChangeAccountbookColorCard, InviteAccountbookCard 파일 생성

- ChangeAccountbookColorCard 파일 생성
- InviteAccountbookCard 파일 생성

JunYoung7 committed:
feat : InviteAccountbookCard 구현

- 초대하기 영역을 만들기 위한 InviteAccountbookCard 구현
- SquircleShortButton, UserImages, SquircleCard를 재사용하여 구현

JunYoung7 committed:
refactor : ChangeAccountbookColorCard stories 수정

- storybook 분류 오류 해결을 위한 organisms에 분류되어있던 것을 molecules로 수정

JunYoung7 committed:
feat : CreateAccountbookColorCard 구현

- 가계부 색상 변화를 위한 CreateAccountbookColorCard 구현
- CreateAccountbookPage에 해당 organisms 추가
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : CreateAccountbookFromBox 수정 

- 시각적 UI 개선을 위해 겉을 감싸는 부분을 Container로 지정하지 않고 SquircleCard로써 구현
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : CreateAccountbookPage style 수정

- ui 개선을 위한 width, margin, height 값 수정
- 보라색 사용을 위한 themes.color.primary.purple 추가
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

JunYoung7 committed:
refactor : TopNavBar Props 적용

- background-color 사용을 위한 TopNavBar Props 적용

JunYoung7 committed:
refactor : 가계부 생성 버튼 수정

- 검색 -> 생성으로 수정

do02reen24 commented on client/src/components/organisms/TopNavBar/TopNavBar.tsx:
```diff
@@ -26,9 +34,9 @@ const FlexRowContainer = styled.div`
+    <FlexRowBox backgroundColor={backgroundColor}>
       <HomeButton />
       <FlexRowContainer>
         <NavButton
```
TopNavBar에 넘겨받은 backgroundColor 를 판별하여 하위 컴포넌트들의 색깔도 흰색 or 갈색(검정)으로 변경하는 것도 추가해주실 수 있으신가요?

JunYoung7 replied:
이 부분 관련해서 문제가 하나있어서 저녁에 잠깐 말씀 드리고 반영해보도록 하겠습니다

do02reen24 commented on client/src/views/CreateAccountbookPage.tsx:
```diff
@@ -1,12 +1,27 @@
       <ColoredBackground backgroundColor={myColor.primary.lightGray} />
+      <CenterContent>
+        <TopNavBar backgroundColor={myColor.primary.purple} />
+        <CreateAccountbookFormBox backgroundColor={myColor.primary.purple} />
```
이 부분은 사실 purple이 아니라 랜덤으로 색상을 받아왔던 거 같아요!
엄청 급한건 아니여서 이번주 남은 작업 보시고 판단해서 반영해주세요! 바쁘시면 제가 콜리 때 개발하겠습니다😀
```js
export const getRandomColor = () => {
  const color = `#${Math.round(Math.random() * 0xffffff).toString(16)}`;
  return color;
};
```
참고로 랜덤으로 색 뽑아내는 함수 남겨둡니다 ㅎㅎ 필요하시면 `utils/color.js` 에 추가해주세요!

JunYoung7 replied:
제가 이번주에 남은 작업 보고 반영하겠습니다 ! 함수 감사합니다 😃 => 반영했습니다

do02reen24 commented on client/src/components/organisms/CreateAccountbookFormBox/CreateAccountbookFormBox.tsx:
```diff
@@ -0,0 +1,47 @@
+        alignItems="center"
+      >
+        <RowFlexContainer width="100%" justifyContent="space-between">
+          <LeftLargeText color="white" fontWeight="bold">
```
이 부분은 받아온 배경색에 따라 white 또는 black을 넘겨주도록 구현해야할 것 같아요!
utils에 color 폴더 만들면 좋을 것 같습니다 ㅎㅎ
```js
export const getFontColor = (color) => {
  const redValue = parseInt(color.substring(1, 3), 16);
  const greenValue = parseInt(color.substring(3, 5), 16);
  const blueValue = parseInt(color.substring(5, 7), 16);
  const result =
    (redValue * 0.299 + greenValue * 0.587 + blueValue * 0.114) / 255;
  if (result >= 0.5) return 'black';
  else return 'white';
};
```
이건 제가 옛날에 만들어둔 함수인데 `#AB12CD` 등의 값을 넘겨주면 됩니다

JunYoung7 replied:
넵 피드백 감사합니다 반영했습니다!

do02reen24 reviewed (commented):
스토리북 만드실 때 항상 control을 꼼꼼하게 설정해주셔서 좋은 것 같습니다👍
수고하셨어요~😄👍🔥

Zigje9 reviewed (approved):
배치가 힘들었을것 같은데 고생하셨습니다!!

leejiwoo2002 reviewed (approved):
고생하셨습니다

JunYoung7 committed:
feat : util에 color 관련 함수 추가 

# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명

- getFontColor, getRandomColor 함수 추가

JunYoung7 committed:
refactor : feadback 반영

-  CreateAccountPage에 background 색상 랜덤으로 넘겨주게 수정
- CreateAccountbookFormBox에서 background 색상에 따라 하위 컴포넌트 색깔 수정
# <Why>를 위한 <What>
# 로그인을 위한 OAuth api 구현
# 객체 의존성 역전을 위한 프로토콜 추상화

# <타입> 리스트
#   feat    : 기능 (새로운 기능)
#   fix     : 버그 (버그 수정)
#   refactor: 리팩토링
#   style   : 스타일 (코드 형식, 세미콜론 추가: 비즈니스 로직에 변경 없음)
#   docs    : 문서 (문서 추가, 수정, 삭제)
#   test    : 테스트 (테스트 코드 추가, 수정, 삭제: 비즈니스 로직에 변경 없음)
#   chore   : 기타 변경사항 (빌드 스크립트 수정 등)
# --------------------------------------------------------------
# 메세지는 한글로 작성
# 타입 첫 문자는 소문자로 작성
# 제목의 끝은 명사형으로 작성
# 제목 끝에 마침표(.) 금지
# 본문은 "어떻게" 보다 "무엇을", "왜"를 설명
